### PR TITLE
Payment profiles

### PR DESCRIPTION
**Pull request by LayneHaber**
## The Problem
Need a way to set expected collateral needs for a group of user addresses.

## The Solution
Added the MVPaymentProfile. Channels can be linked to a `payment_profile_id` by an admin, and the parameters are stored in a new `payment_profiles` table.

Key parts of the code will be migration and dao related, but it is an admin only endpoint at the moment so we should be okay.

## Checklist:

- [x] I have incremented the version in the project root's package.json
- [x] The commit that increments the version includes the new version number in it's commit message
- [x] My code follows the code style of this project.
- [x] This code has passed all CI tests & has been deployed successfully to staging.
- [x] I have verified that, the staging site reflected these changes and worked as expected.
- [x] I have described any backwards-incompatibility implications above.
- [x] I have highlighted which parts of the code should be reviewed most carefully.
- [x] If my change requires a change to the documentation, I have updated it accordingly.

### CODE REVIEW

File: modules/database/migrations/sqls/20190507194131-add-payment-profiles-up.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+/* Simple payment profile with min collateral and collateral target */
+create table payment_profiles (
+	id bigserial primary key,
+  minimum_maintained_collateral_wei wei_amount default '0',
```
Comment (rhlsthrm) [inline modules/database/migrations/sqls/20190507194131-add-payment-profiles-up.sql]: Minor nitpick, if you're going to default it like this, you might as well make it `not null`.

File: modules/database/migrations/sqls/20190507194131-add-payment-profiles-up.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  amount_to_collateralize_wei wei_amount default '0',
+  amount_to_collateralize_token wei_amount default '0'
+);
+
+/* Add payment profile id to the channels column */
+alter table _cm_channels add column payment_profile_id bigint null;
+alter table _cm_channels add constraint payment_profile_id_fk foreign key (payment_profile_id) references payment_profiles(id);
+
+/* Update the channels view */
+create or replace view cm_channels as (select * from _cm_channels);
```
Comment (rhlsthrm) [inline modules/database/migrations/sqls/20190507194131-add-payment-profiles-up.sql]: We probably honestly shouldn't be doing this, we should only put what we need/use in the view but let's worry about that later.

File: modules/hub/src/PaymentProfilesService.test.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+const {
+  toWeiString,
+} = big
+
+describe('PaymentProfilesService', () => {
+  const registry = getTestRegistry()
+  const service = registry.get('PaymentProfilesService')
+
+  // **** helper functions
```
Comment (rhlsthrm) [inline modules/hub/src/PaymentProfilesService.test.ts]: Very nice.

File: modules/hub/src/api/PaymentProfilesApiService.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    if (!config) {
+      res.status(400)
+      res.send({ error: `No payment profile config found with id: ${id}` })
+    }
+
+    return res.send(config)
+  }
+
+  async doGetPaymentProfileByUser(req: express.Request, res: express.Response) {
+    if (!isAdmin(req)) {
```
Comment (rhlsthrm) [inline modules/hub/src/api/PaymentProfilesApiService.ts]: I think it's fine to make this `ownedAddressOrAdmin`, users should be able to see their own profile I think.

Comment (LayneHaber) [reply modules/hub/src/api/PaymentProfilesApiService.ts]: Made it with user auth, since we specd it out client side

**Pull request merged**